lightning/src/routing/network_graph.rs

```diff
@@ -1857,8 +1940,26 @@ mod tests {
 			};
 		}
 
-		// Permanent closing deletes a channel
-		{
+		if remove_by_timeout {
+			network_graph.remove_stale_channels_with_time(100 + 60 * 60 * 24 * 14);
+			assert_eq!(network_graph.read_only().channels().len(), 1);
+			assert_eq!(network_graph.read_only().nodes().len(), 2);
+
+			network_graph.remove_stale_channels_with_time(101 + 60 * 60 * 24 * 14);
+			#[cfg(feature = "std")]
+			{
+				// In std mode, a further check is performed before fully removing the channel -
+				// the channel_announcement must have been received at least two weeks ago. We
+				// fudge that here by indicating the time has jumped two weeks.
+				assert_eq!(network_graph.read_only().channels().len(), 1);
+				assert_eq!(network_graph.read_only().nodes().len(), 2);
```

Comment:
Would this mean the checks at the end of the test will fail when `std` feature is not set?

Reply:
They shouldnt?